src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1967,11 +1967,12 @@ def list_locations(cmd, sku, linux_workers_enabled=None):
     return client.list_geo_regions(full_sku, linux_workers_enabled)
 
 
-def _check_zip_deployment_status(deployment_status_url, authorization):
+def _check_zip_deployment_status(deployment_status_url, authorization, timeout=None):
     import requests
     import time
     num_trials = 1
-    while num_trials < 10:
+    total_trials = (int(timeout) // 30) if timeout else 10
```

Comment:
Based on the `time.sleep(15)` below, I believe the default is now at 15*10 seconds which is two and half minutes. Should we go with `20`? Also why not just set the default value in the function argument declarations ?

Reply:
thanks for pointing this out, i had updated the sleep to 30 so that the default is 10*30.

Reply:
You are very welcome.
One more thing, I believe the `num_trials ` should start with 0; otherwise, you only wait 9 times